chef/cookbooks/postgresql/recipes/server.rb

```diff
@@ -98,14 +98,57 @@
 
 change_notify = node["postgresql"]["server"]["config_change_notify"]
 
+raw_configuration = node["postgresql"]["config"]
+postgresql_configuration = {}
+raw_configuration.each do |key, value|
+  next if value.nil?
+  formatted_value = case value
+  when String
+    "'#{value}'"
+  when TrueClass
+    "on"
+  when FalseClass
+    "off"
+  else
+    value
+  end
+  postgresql_configuration[key] = formatted_value
+end
+
+reloadable_parameters = [
+  "log_line_prefix"
+]
+
+config_needs_restart = postgresql_configuration.reject do |key, value|
+  reloadable_parameters.include? key
+end
+
+config_needs_reload = postgresql_configuration.select do |key, value|
+  reloadable_parameters.include? key
+end
+
 template "#{node['postgresql']['dir']}/postgresql.conf" do
   source "postgresql.conf.erb"
   owner "postgres"
   group "postgres"
   mode 0600
+  variables(
+    config: config_needs_restart.sort
+  )
   notifies change_notify, "service[postgresql]", :immediately
```

Comment:
This is causing an issue for me, it seems that this causes the service to be immediately reloaded after the template change, but before the postgresql_reloadable.conf exists, so postgresql fails due to a syntax error because the incuded file doesn't exist yet.